File: backend/verify_bathrooms.js
const BASE_URL = "http://localhost:8000";

async function verifyBathrooms() {
    console.log("Starting verification...");

    // 1. Create a flat with 3 bathrooms
    console.log("Step 1: Creating flat (bathrooms=3)...");
    const uniqueSuffix = Date.now();
    const flatNumber = `BATH_${uniqueSuffix}`;
    const formData = new FormData();
    formData.append("flat_number", flatNumber);
    formData.append("floor_number", "1");
    formData.append("bedrooms", "2");
    formData.append("bathrooms", "3");
    formData.append("address", "Test Bath Building JS");

    try {
        const createRes = await fetch(`${BASE_URL}/flats`, {
            method: "POST",
            body: formData,
        });

        if (!createRes.ok) {
            console.error("Create failed:", await createRes.text());
            return;
        }

        const createData = await createRes.json();
        console.log("Create response:", createData); // Debug log
        const flatUuid = createData.uuid;
        console.log("Flat created with UUID:", flatUuid);

        if (createData.bathrooms === 3) {
            console.log("[SUCCESS] Flat created with bathrooms=3");
        } else {
            console.error(`[FAILED] Flat created with bathrooms=${createData.bathrooms}`);
            return;
        }

        // 2. Verify GET /properties
        console.log("Step 2: Verifying GET /properties...");
        const propsRes = await fetch(`${BASE_URL}/properties`);
        const props = await propsRes.json();
        const targetProp = props.find((p) => p.uuid === flatUuid);

        if (targetProp && targetProp.bathrooms === 3) {
            console.log("[SUCCESS] GET /properties includes correct bathrooms count");
        } else {
            console.error(`[FAILED] GET /properties mismatch. Found:`, targetProp);
        }

        // 3. Update flat to 4 bathrooms
        console.log("Step 3: Updating flat to 4 bathrooms...");
        const updatePayload = {
            action: "UPDATE_FLAT_ONLY",
            flat_details: {
                bedrooms: 2,
                bathrooms: 4,
                floor_number: 1,
                address: "Test Bath Building JS",
            },
        };

        const updateRes = await fetch(`${BASE_URL}/flats/${flatUuid}`, {
            method: "PATCH",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(updatePayload),
        });

        if (!updateRes.ok) {
            console.error("Update failed:", await updateRes.text());
            return;
        }

        const updateData = await updateRes.json();
        if (updateData.bathrooms === 4) {
            console.log("[SUCCESS] Flat updated to bathrooms=4");
        } else {
            console.error(`[FAILED] Update result mismatch: bathrooms=${updateData.bathrooms}`);
        }

        console.log("\nVERIFICATION COMPLETE: ALL SUCCESS");

    } catch (error) {
        console.error("Verification error:", error);
    }
}

verifyBathrooms();

File: backend/verify_bathrooms.py
import requests
import json
import os

BASE_URL = "http://localhost:8000"

def log(msg, success=None):
    if success is True:
        print(f"[SUCCESS] {msg}")
    elif success is False:
        print(f"[FAILED] {msg}")
    else:
        print(f"[INFO] {msg}")

def verify_bathrooms():
    # 1. Create a flat with 3 bathrooms
    log("Step 1: Creating flat with 3 bathrooms...")
    files = {
        'flat_number': (None, 'BATH_TEST_01'),
        'floor_number': (None, '1'),
        'bedrooms': (None, '2'),
        'bathrooms': (None, '3'),
        'address': (None, 'Test Bath Building')
    }
    
    try:
        res = requests.post(f"{BASE_URL}/flats", files=files)
        if res.status_code != 200:
            log(f"Create failed: {res.text}", False)
            return
        
        flat_data = res.json()["flat"]
        flat_uuid = flat_data["uuid"]
        
        if flat_data.get("bathrooms") == 3:
            log("Flat created with bathrooms=3", True)
        else:
            log(f"Flat created but bathrooms={flat_data.get('bathrooms')}", False)
            return

        # 2. Verify GET /properties
        log("Step 2: Verifying GET /properties...")
        res = requests.get(f"{BASE_URL}/properties")
        props = res.json()
        target_prop = next((p for p in props if p["uuid"] == flat_uuid), None)
        
        if target_prop and target_prop.get("bathrooms") == 3:
            log("GET /properties returns correct bathrooms count", True)
        else:
            log(f"GET /properties failed. Found: {target_prop}", False)

        # 3. Update flat to 4 bathrooms
        log("Step 3: Updating flat to 4 bathrooms...")
        update_payload = {
            "action": "UPDATE_FLAT_ONLY",
            "flat_details": {
                "bedrooms": 2,
                "bathrooms": 4,
                "floor_number": 1,
                "address": "Test Bath Building"
            }
        }
        res = requests.patch(f"{BASE_URL}/flats/{flat_uuid}", json=update_payload)
        
        if res.status_code != 200:
            log(f"Update failed: {res.text}", False)
            return

        updated_flat = res.json()
        if updated_flat.get("bathrooms") == 4:
            log("Flat updated to bathrooms=4", True)
        else:
            log(f"Update returned bathrooms={updated_flat.get('bathrooms')}", False)

        # 4. Cleanup (optional, but good practice if delete endpoint existed)
        # For now we just leave it.

    except Exception as e:
        log(f"Exception during verification: {e}", False)

if __name__ == "__main__":
    verify_bathrooms()
